src/main/java/TurnUseCases/TradeUseCase/TradeOffer.java

```diff
@@ -0,0 +1,153 @@
+    private boolean isValid;
+
+    /**
+     * Creates a new TradeOffer.
+     *
+     * @param tradeMoney if positive, it is the amount of money player1 is offering player2.
+     *                   If negative, it is the amount of money player1 wants from player2.
+     * @param jailCard if 1, then player1 is offering a get out of jail card to player2.
+     *                 If -1, then player1 is requesting a get out of jail card from player2.
+     *                 If 0, then no get out of jail cards are offered or requested.
+     * @param propertiesOffered the list of properties player1 is offering player2.
+     * @param propertiesReceived the list of properties player1 wants from player2.
+     * @param player1 the player who made the trade offer.
+     * @param player2 the player who is receiving the trade offer.
+     */
+    public TradeOffer(int tradeMoney, int jailCard,
+                      ArrayList<Property> propertiesOffered, ArrayList<Property> propertiesReceived,
+                      Player player1, Player player2){
+        this.tradeMoney = tradeMoney;
+        this.jailCard = jailCard;
+        this.propertiesOffered = propertiesOffered;
+        this.propertiesReceived = propertiesReceived;
+        this.player1 = player1;
+        this.player2 = player2;
+        this.isValid = CheckIsValid();
+
+    }
+
+    /**
+     * Checks if this TradeOffer is valid or not.
+     *
+     * @return Whether this TradeOffer is valid or not.
+     */
+    public boolean CheckIsValid(){
+        if(tradeMoney > 0 && player1.getMoney() < tradeMoney){
+            return false;
+        } else if (tradeMoney < 0 && player2.getMoney() < - tradeMoney) {
+            return false;
+        } else if (!CheckPropertiesOffered()) {
+            return false;
+        } else if (!CheckPropertiesReceived()) {
+            return false;
+        } else {
+            return CheckJailCard();
+        }
+
+    }
+
+    /**
+     * Checks if player1 actually owns all the properties he/she is offering.
+     *
+     * @return whether player1 owns all the properties offered or not.
+     */
+    public boolean CheckPropertiesOffered(){
+        for (Property p : propertiesOffered){
+            if(!player1.getProperties().contains(p)){
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks if player2 actually owns all the properties he/she is offering.
+     *
+     * @return whether player2 owns all the properties offered or not.
+     */
+    public boolean CheckPropertiesReceived(){
+        for (Property p : propertiesReceived){
+            if(!player2.getProperties().contains(p)){
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Checks if the player offering the jail card has one.
+     *
+     * @return whether the player offering the jail card has one.
+     */
+    public boolean CheckJailCard(){
```

Comment:
the player class does have a method that will return true if the player owns a get out of jail free card

Reply:
its called hasGetOutofJailFreeCard